scripts/sct_create_mask.py

```diff
@@ -284,70 +276,67 @@ def get_parser():
-                                  '  centerline: volume that contains centerline. E.g.: centerline,my_centerline.nii',
-                      mandatory=False,
-                      deprecated_by='-p')
-    parser.add_option(name='-size',
-                      type_value='str',
-                      description='Size of the mask in the axial plane, given in pixel (ex: 35) or in millimeter (ex: 35mm). If shape=gaussian, size corresponds to "sigma"',
-                      mandatory=False,
-                      default_value=param_default.size,
-                      example=['45'])
-    parser.add_option(name='-s',
-                      type_value=None,
-                      description='Size in voxel. Odd values are better (for mask to be symmetrical). If shape=gaussian, size corresponds to "sigma"',
-                      mandatory=False,
-                      deprecated_by='-size')
-    parser.add_option(name='-f',
-                      type_value='multiple_choice',
-                      description='Shape of the mask.',
-                      mandatory=False,
-                      default_value=param_default.shape,
-                      example=param_default.shape_list)
-    parser.add_option(name='-o',
-                      type_value='file_output',
-                      description='Name of output mask.',
-                      mandatory=False,
-                      example=['data.nii'])
-    parser.add_option(name="-r",
-                      type_value="multiple_choice",
-                      description='Remove temporary files.',
-                      mandatory=False,
-                      default_value='1',
-                      example=['0', '1'])
-    parser.add_option(name="-v",
-                      type_value='multiple_choice',
-                      description="verbose: 0 = nothing, 1 = classic, 2 = expended",
-                      mandatory=False,
-                      example=['0', '1', '2'],
-                      default_value='1')
+
+    parser = argparse.ArgumentParser(description='Create mask along z direction.',
+                                     add_help=None,
+                                     prog=os.path.basename(__file__).strip(".py")
+                                     )
+    mandatoryArguments = parser.add_argument_group("\nMandatory arguments")
+    mandatoryArguments.add_argument('-i',
+                        help='Image to create mask on. Only used to get header. Must be 3D, e.g. \'data.nii.gz\')',
+                        required = True
+                        )
+    mandatoryArguments.add_argument('-p',
+                        help='Process to generate mask.\n'
+                        'coord: X,Y coordinate of center of mask. e.g. \'coord,20x15\'\n'
+                        'point: volume that contains a single point. e.g. \'point,label.nii.gz\'\n'
+                        'center: mask is created at center of FOV.\n'
```

Comment:
@YangHee-Min : (I remember that it has been mentioned during the last SCT meeting, but can't remember the answer) --> Would it be possible to add a "return" here?
Note that I can see the `\n` in the code but herebelow my terminal output:
```
  -p P                  Process to generate mask. coord: X,Y coordinate of
                        center of mask. e.g. 'coord,20x15' point: volume that
                        contains a single point. e.g. 'point,label.nii.gz'
                        center: mask is created at center of FOV. centerline:
                        volume that contains centerline or segmentation. (e.g.
                        'centerline,t2_seg.nii.gz')
```

Reply:
It can be done using `formatter_class = argparse.RawDescriptionHelpFormatter`, but be aware that it might change the way argument help text is displayed

Reply:
@YangHee-Min can you show us an example when using this formatter class? Other function help (e.g. register_multimodal) are much more complex than this one so we will definitely end up needing proper formatting for some parameters.

Reply:
[(Here)](https://github.com/neuropoly/spinalcordtoolbox/blob/48514eee05c3b32dca3394bf19995461c6a7082a/scripts/sct_analyze_lesion.py#L42) is an example with `sct_analyze_lesion` and [(here) ](https://pastebin.com/U6bcpRWf) is the output

Reply:
@YangHee-Min this example (already implemented) is for the description, not the help specific to a flag (which is what we are discussing here)

Reply:
Oh sorry about that! I gave a bad example for it but as the link [(here)](https://stackoverflow.com/questions/3853722/python-argparse-how-to-insert-newline-in-the-help-text/22157136) suggests, the formatter class is applied onto the the arguments as well if I understood the proposed solution in the link correctly

Reply:
ok, cool. Can we see how the formatting would look like when applied to a flag descrption (not the general description)? thanks

Reply:
Sure! [(Here)](https://pastebin.com/72krH0pL) is how the the output looks with `formatter_class= argparse.RawTextHelpFormatter` added (`-p` argument is on line 14). If you want the new code without the other changes to be added (just formatter_class added), you can find it [(here)](https://github.com/neuropoly/spinalcordtoolbox/blob/455a431f8e1b217e1d0e31ad65936b601c28a82e/scripts/sct_create_mask.py#L283)

Reply:
ok that looks great! i would just indent (and use same convention for e.g.):
~~~
  -p P                  Process to generate mask:
                          <coord,XxY>: Center mask at the X,Y coordinates. Example: "coord,20x15"
                          <point,FILE>: Center mask at the X,Y coordinates of the label defined in input volume FILE. Example: "point,label.nii.gz"
                          <center>: Center mask in the middle of the FOV (nx/2, ny/2).
                          <centerline,FILE>: At each slice, the mask is centered at the spinal cord centerline, defined by the input segmentation FILE. Example: "centerline,t2_seg.nii.gz"
~~~

Reply:
Is [this (commit 64269d9)](https://github.com/neuropoly/spinalcordtoolbox/commit/64269d9979757dcbd2a354e8c87babb58be4b07e) what you wanted? With this you get the following output: https://pastebin.com/ns4vvpB3 
(note that in terminal the lines with the example that aren't indented are supposed to continue)